infrastructure-playbooks/shrink-mgr.yml

```
@@ -0,0 +1,116 @@
+---
+# This playbook shrinks the Ceph manager from your cluster
+#
+# Use it like this:
+# ansible-playbook shrink-mgr.yml -e mgr_to_kill=ceph-mgr1
+#     Prompts for confirmation to shrink, defaults to no and
+#     doesn't shrink the cluster and yes shrinks the cluster.
+#
+# ansible-playbook -e ireallymeanit=yes|no shrink-mgr.yml
+#     Overrides the prompt using -e option. Can be used in
+#     automation scripts to avoid interactive prompt.
+
+
+- name: gather facts and check the init system
+  hosts:
+    - "{{ mon_group_name | default('mons') }}"
+    - "{{ mgr_group_name | default('mgrs') }}"
+  become: true
+  tasks:
+    - debug: msg="gather facts on all Ceph hosts for following reference"
+
+- name: confirm if user really meant to remove manager from the ceph cluster
+  hosts: localhost
+  become: true
+  vars_prompt:
+    - name: ireallymeanit
+      prompt: Are you sure you want to shrink the cluster?
+      default: 'no'
+      private: no
+  vars:
+    mon_group_name: mons
+    mgr_group_name: mgrs
+
+  pre_tasks:
+    - name: exit playbook, if only one manager is present in cluster
+      fail:
+        msg: "You are about to shrink the only manager present in the cluster."
+      when: groups[mgr_group_name] | length | int == 1
```

even if 0 mgr isn't a valid deployment, it would pass this task

```suggestion
      when: groups[mgr_group_name] | length | int < 2
```

---

done.